src/main/java/org/example/bookstore/service/impl/BookServiceImpl.java

```diff
@@ -35,4 +35,26 @@ public BookDto findById(Long id) {
         return bookOptional.map(bookMapper::toDto).orElseThrow(() -> new EntityNotFoundException(
                 "Book with %d not found".formatted(id)));
     }
+
+    @Override
+    public BookDto update(Long id, CreateBookRequestDto requestDto) {
+        checkIfBookExists(id);
+        Book book = bookMapper.toModel(requestDto);
+        book.setId(id);
+        return bookMapper.toDto(bookRepository.save(book));
+    }
+
+    @Override
+    public void deleteById(Long id) {
+        checkIfBookExists(id);
+        bookRepository.findById(id).orElseThrow(() -> new EntityNotFoundException(
+                "Book with %d not found".formatted(id)));
```

Comment:
duplicated existence check
```suggestion
        checkIfBookExists(id);
```